### PR TITLE
Subschema valildation

### PR DESCRIPTION
**Pull request by AliciaSchep**
A PR to add support for validating a subschema #18 

Does not yet add a section to vignette -- the vignette does not look like it's been updated yet to account for the 'ajv' option, so I was wondering if it made sense to split up into a vignette for each engine? or whether it might make sense to introduce that as well in these changes? Not sure if 1 comprehensive versus several smaller vignettes is desired approach @richfitz ?  E.g. I could imagine a small vignette 'Engine choice: imjv versus ajv', and another 'validating subschemas' that are specific to those topics.

### CODE REVIEW

File: R/validate.R [hunk 1]
```diff
@@ -10,14 +10,25 @@
...
+##'   would check that the json is a valid "Hello" object. Only available if 
+##'   \code{engine = 'ajv'}.
+##'   
 ##' @export
 ##' @example man-roxygen/example-json_validator.R
-json_validator <- function(schema, engine = "imjv") {
+json_validator <- function(schema, engine = "imjv", reference = NULL) {
   schema <- get_string(schema)
+  if (!is.null(reference) && engine != 'ajv') {
+    # Should this instead use ajv and give a warning?
```
Comment (richfitz) [inline R/validate.R]: No, a stop here is the right behaviour (you can remove this comment)

File: R/validate.R [hunk 2]
```diff
@@ -75,10 +86,14 @@ json_validator_ajv <- function(schema) {
     `draft-06` = "ajv",
     `draft-07` = "ajv",
   )
+  
+  schema_name <- basename(tempfile("schema_"))
+  if (is.null(reference)) reference <- schema_name
```
Comment (richfitz) [inline R/validate.R]: Can you please reformat over 3 lines with curly braces?

**Comment by richfitz** [comment]
(sorry, doing some refactoring work atm - see #21 - and would like to incorporate your changes)

**Comment by AliciaSchep** [comment]
Sorry meant to get to these changes but have been swamped this week! Happy to make a separate PR later to add to the vignette.

**Comment by richfitz** [comment]
No worries at all - thanks for the PR!  I'll try and do some work on the vignette in the next week or two, but if you would like to tackle the changes there I'd be happy to accept that

**Pull request merged**